crates/viewer/re_space_view_dataframe/src/view_query_v2/ui.rs

```diff
@@ -0,0 +1,532 @@
+                .filter_map(|c| {
+                    c.indicator_component_archetype()
+                        .and_then(|archetype_short_name| {
+                            ctx.reflection
+                                .archetype_reflection_from_short_name(&archetype_short_name)
+                        })
+                })
+                .flat_map(|archetype_reflection| {
+                    archetype_reflection
+                        .required_fields()
+                        .map(|field| field.component_name)
+                })
+                .filter(|c| all_components.contains(c))
+                .collect::<ComponentNameSet>()
+        };
+
+        // If the currently saved component, we auto-switch it to a reasonable one.
+        let mut event_component = event_component
+            .and_then(|component| all_components.contains(&component).then_some(component))
+            .or_else(|| suggested_components().first().copied())
+            .unwrap_or_else(|| ComponentName::from("-"));
+
+        //
+        // UI for event entity and component
+        //
+
+        ui.add_enabled_ui(active, |ui| {
+            ui.spacing_mut().item_spacing.y = 0.0;
+
+            ui.list_item_flat_noninteractive(list_item::PropertyContent::new("Entity").value_fn(
+                |ui, _| {
+                    egui::ComboBox::new("pov_entity", "")
+                        .selected_text(event_entity.to_string())
+                        .show_ui(ui, |ui| {
+                            for entity in all_entities {
+                                let label = entity.to_string();
+                                ui.selectable_value(&mut event_entity, entity, label);
+                            }
+                        });
+                },
+            ));
+
+            ui.list_item_flat_noninteractive(
+                list_item::PropertyContent::new("Component").value_fn(|ui, _| {
+                    egui::ComboBox::new("pov_component", "")
+                        .selected_text(event_component.short_name())
+                        .show_ui(ui, |ui| {
+                            for component in all_components {
+                                let label = component.short_name();
+                                ui.selectable_value(&mut event_component, component, label);
+                            }
+                        });
+                }),
+            );
+        });
+
+        //
+        // Save event if changed
+        //
+
+        let filter_by_event =
+            components::FilterByEvent::new(active, &event_entity, event_component);
+
+        if original_filter_by_event.as_ref() != Some(&filter_by_event) {
+            self.save_filter_by_event(ctx, &filter_by_event);
+        }
+
+        Ok(())
+    }
+
+    pub(super) fn column_visibility_ui(
+        &self,
+        ctx: &ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        timeline: &Timeline,
+        schema: &[ColumnDescriptor],
```

Comment:
I really don't like the name `schema` (here and everywhere else) in the context of the new APIs.

We need to be explicit, always, or it's extremely hard to follow what's going on: is this the `view_columns` (which control the view contents) or the `selected_columns` (which control the selection)?

Reply:
That make sense to me.

`schema` -> always arrow schema
`view_columns` -> all columns in the view (depends on the `view_contents`)
`selected_columns` -> the columns that remains after applying the user selection (`select` arg/func in the API)

This feels like something @jleibs should be on-board with though.

I'm applying some of that here, and will hunt every remaining `schema` in part 6.